src/components/PanelV2/PanelV2.js

```diff
@@ -119,47 +112,50 @@ export default class PanelV2 extends Component {
 
     return (
       <Transition className={namespace}>
-        {this.state.isOpen && (
+        {isOpen && (
           <Portal
             focusTrap={focusTrap}
             stopPropagation={stopPropagation}
             stopPropagationEvents={stopPropagationEvents}
+            onOverlayClick={onClose}
           >
+            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
             <section
               className={classnames(namespace, className)}
               role="dialog"
               aria-label={ariaLabel}
               aria-modal="true"
+              onKeyDown={handleKeyDown}
```

Comment:
One thing I've noticed here that is potentially problematic is that if a user clicks a non-interactive element of the `Panel` while it's open, for example selecting the text, the `onKeyDown` handler won't trigger:

![Panel](https://user-images.githubusercontent.com/3846874/84380441-41e7fa00-abdf-11ea-9cdf-693470c5c4d9.png)

Reply:
Good catch! Looking into this, I had to add `tabIndex={-1}`. 👀 

From MDN:

> Keyboard events are only generated by <inputs>, <textarea> and anything with the contentEditable attribute or with tabindex="-1".

https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event

The `Modal` appears to take the same approach with `tabIndex`, etc.